src/StateSpace.hpp

```diff
@@ -23,11 +23,18 @@ class StateSpace {
      * Finds a state in the direction of @target from @source.state().
      * This new state will potentially be added to the tree.  No need to do
      * any validation on the state before returning, the tree will handle
-     * that.
+     * that. A negative stepSize tells it to implement adaptive stepsize control.
      */
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control.
+     *
+     * @param ascLimit The maximum size that the stepsize can be, even when there are no nearby obstacles
+     */
+    virtual T intermediateState(const T &source, const T &target, float stepSize, float ascLimit) const = 0;
```

Comment:
Same comment as before, it may be a little easier to read if you name them minStepSize and maxStepSize.